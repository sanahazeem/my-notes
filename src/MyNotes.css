.notes-app {
  display: flex;
  flex-basis: 100%;
  align-content: flex-start;
  justify-content: center;
  background-color:rgb(157, 164, 204);
  flex-wrap: wrap;
  height: 150vh;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.notes-app .header {
  display: flex;
  flex-basis: 100%;
  justify-content: center;
  align-items: center;
  height: 180px;
  color:  rgb(73, 93, 207);
  background-color:  rgb(73, 93, 207);
}

.notes-app .title-box{
  flex-basis: 600px;
  text-align: center;
  background-color:  rgb(157, 164, 204);
  border: 1px solid #fff;
  font-size: 25px;
  font-weight: 700;
}

.notes-app .notes-text-area {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: row;
  flex-basis: 100%;
}

.btn-container{
  display: flex;
  flex-basis: 100%;
  justify-content: center;
  align-items: center;
  padding: 20px 0;
  padding-top: 40px
}

.note-btn {
  flex-basis: 600px;
  padding: 10px 20px;
  font-size: 25px;
  font-weight: 700;
  letter-spacing: 2px;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  color:  rgb(73, 93, 207);
  border-radius: 2px;
  cursor: pointer;
}

.delete-btn-container {
  display: flex;
  flex-basis: 100%;
  padding: 20px 5px 0px;
}

.delete-btn {
  flex-basis: 100px;
  padding: 10px;
  background-color: #fff;
  color:  rgb(73, 93, 207);
  font-weight: 700;
  letter-spacing: 2px;
  font-size: 15px;
  cursor: pointer;
}

.notes-app .note-box {
  flex-basis: 40%;
  margin: 10px 15px;
  box-shadow: 0 0 40px 0 rgba(0, 0, 0, 0.1);
}

.notes-app .form-box {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 20px;
  background-color: rgb(73, 93, 207);
}

.notes-app .label {
  width: 100%;
  margin: 10px 0;
  font-size: 25px;
  color: #fff;
  font-weight: 700;
}

.notes-app .text-area {
  width: 100%;
  height: 130px;
   font-size: 15px;
}

