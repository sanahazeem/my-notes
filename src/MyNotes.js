import React, { Component }  from 'react';
import './MyNotes.css';

class MyNotes extends Component {
  state={
    notes: []
  };

  addNote = () =>  {
    const {notes} = this.state;
    const ids = notes.map(({id})=> id);
    const notesCopy= [...this.state.notes];
    const max_id =ids.length > 0 ? Math.max(...ids): 0;
    notesCopy.push({id: max_id + 1});
    this.setState({notes: notesCopy});
  };

  removeNote = ({id: noteId}) => {
    const {notes} = this.state
    const Note = notes.filter(({id}) => id!== noteId);
    this.setState({notes: Note});
  };

  render () {
    const {notes} = this.state;
    return (
      <div className="notes-app">
      <header className="header">
        <div className="title-box">
          <h1 className="title">My Notes App</h1>
        </div>
      </header>
      <div className="btn-container">
        <button type="button" onClick={() => this.addNote()} className="note-btn add-btn">
          Click here to create a note :)
        </button>

      </div>
      <section className="notes-text-area">
        {
          notes.map (({id}) => {
              return (
                  <div className="note-box" key={id}>
                    <div className="form-box">
                    <label className="label">Note</label>
                    <textarea className="text-area" id="textarea1" placeholder="Add a new note!" class="text-area"/>
                      <div className="delete-btn-container">
                    <button onClick={()=>this.removeNote({id})} className="delete-btn">Delete</button>
                    </div>
                    </div>
                  </div>
              )
          })
      }
    </section>
    </div>
    );
  };

};
export default MyNotes;
