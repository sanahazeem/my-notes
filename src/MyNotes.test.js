import React from 'react';
import { shallow } from 'enzyme';
import MyNotes from './MyNotes';


describe ('<MyNotes/>', () => {
    const MainWrapper = shallow (<MyNotes/>)

    it('checks if <MyNotes/> renders correctly', () => {
        expect(MainWrapper).toMatchSnapshot();
    });

    it('checks if the title renders and contains same value', () => {
        expect(MainWrapper.find('.title').text()).toEqual('My Notes App');
    })

    describe('when clicking the `addNotes` btn', () => {
        const id = 1;
        beforeEach (() => {
            MainWrapper.find('.add-btn').simulate('click');
        });

        afterEach (() => {
            MainWrapper.setState({notes:[]})
        });

        it ('adds a new note to the `state`', () => {
            expect(MainWrapper.state().notes).toEqual([{id}])
        });

        it ('adds a new note to the rendered ', () => {
            expect(MainWrapper.find('.notes-text-area').children().length).toEqual(1);
        });

        describe ('when the user wants to remove the added note', () => {
            beforeEach(()=>{
                MainWrapper.instance().removeNote({id})
            })
            it ('removes the note from `state`', () => {
                expect(MainWrapper.state().notes).toEqual([]);
            })
        })
    });


});