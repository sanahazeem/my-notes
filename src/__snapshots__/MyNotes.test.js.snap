// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyNotes/> checks if <MyNotes/> renders correctly 1`] = `ShallowWrapper {}`;
